src/main/groovy/com/github/jk1/license/render/LicenseDataCollector.groovy

```diff
@@ -0,0 +1,79 @@
+package com.github.jk1.license.render
+
+import com.github.jk1.license.License
+import com.github.jk1.license.LicenseReportPlugin
+import com.github.jk1.license.ModuleData
+
+class LicenseDataCollector {
+
+    protected static List<String> singleModuleLicenseInfo(LicenseReportPlugin.LicenseReportExtension config, ModuleData data) {
+        def info = multiModuleLicenseInfo(config, data, false)
+        def moduleUrl = info.moduleUrls?.last()
+        def license = info.licenses?.last()
+        def moduleLicense = license?.name
+        def moduleLicenseUrl = license?.url
+
+        [moduleUrl, moduleLicense, moduleLicenseUrl]
+    }
+
+    protected static MultiLicenseInfo multiModuleLicenseInfo(
+        LicenseReportPlugin.LicenseReportExtension config, ModuleData data, boolean alwaysCheckLicenseFiles = true) {
+        MultiLicenseInfo info = new MultiLicenseInfo()
+
+        data.manifests.each {
+            if (it.url) {
+                info.moduleUrls << it.url
+            }
+            if (it.license) {
+                if (it.license.startsWith('http')) {
+                    info.licenses << new License(url: it.license)
+                } else {
+                    info.licenses << new License(name: it.license)
+                }
+            }
+        }
+        data.poms.each {
+            if (it.projectUrl) {
+                info.moduleUrls << it.projectUrl
+            }
+            if (it.licenses) {
+                info.licenses.addAll(it.licenses)
+            }
+        }
+
+        // check just here for backward compatibility reason of "simple" json-renderer
+        // think about removing this at all and risk, that the simple-renderers gets a different result in case
```

Comment:
Nitpick: you don't want a comma after "risk". You could (optionally) put one after "at all", though.

Reply:
:) roger that